core/src/main/java/com/linecorp/armeria/server/Http2RequestDecoder.java

```diff
@@ -244,7 +244,11 @@ private boolean handle100Continue(int streamId, Http2Headers headers) {
 
         // Send a '100 Continue' response.
         assert encoder != null;
-        encoder.writeHeaders(0 /* unused */, streamId, CONTINUE_RESPONSE, false);
+        final HttpMethod method = HttpMethod.tryParse(methodText.toString());
+        if (method == null) {
+            return false;
```

Comment:
How about handling this at line 111? I think this isn't invalid expect header. 😉